src/main/webapp/app/exercises/modeling/assess/modeling-assessment.component.ts

```diff
@@ -270,9 +281,27 @@ export class ModelingAssessmentComponent implements AfterViewInit, OnDestroy, On
             elementType: feedback.referenceType! as UMLElementType | UMLRelationshipType,
             score: feedback.credits!,
             feedback: feedback.text || undefined,
+            label: this.calculateLabel(feedback),
+            labelColor: this.calculateLabelColor(feedback),
         }));
         if (this.apollonEditor) {
             this.apollonEditor!.model = this.model;
         }
     }
+
+    private calculateLabel(feedback: any) {
+        if (this.highlightDifferences && this.languageService.getCurrentLanguage() === 'en') {
+            return feedback.copiedFeedbackId ? 'First correction round' : 'Second correction round';
+        } else if (this.highlightDifferences && this.languageService.getCurrentLanguage() === 'de') {
+            return feedback.copiedFeedbackId ? 'Erste Korrekturrunde' : 'Zweite Korrekturrunde';
```

Comment:
This seems to work, but also like a hardcoded version of the Artemis translation pipe that we should use instead? 
(I am not sure, that is why I am asking)
![image](https://user-images.githubusercontent.com/65814168/120630588-45ffce00-c467-11eb-8ad5-609a1999b010.png)
If this is ok, when (in which cases) should we use the Artemis translation pipe instead?

Reply:
After some testing I got the pipe to work on typescript files as well, its being used similar to the inline html now, thanks!